sys/include/net/netopt.h

```diff
@@ -227,143 +276,149 @@ typedef enum {
+     * @brief   (uint8_t) device type
+     *
+     * e.g. NETDEV_TYPE_ETHERNET, NETDEV_TYPE_IEEE802154, etc.
      */
     NETOPT_DEVICE_TYPE,
 
     /**
-     * @brief get/set the channel page as defined by IEEE 802.15.4
+     * @brief   (uint8_t) channel page as defined by IEEE 802.15.4
      */
     NETOPT_CHANNEL_PAGE,
 
     /**
-     * @brief get/set the CCA threshold for the radio transceiver
+     * @brief   (int8_t) CCA threshold for the radio transceiver
      *
      * This is the value, in dBm, that the radio transceiver uses to decide
-     * if the channel is clear or not (CCA). If the current signal strength
+     * whether the channel is clear or not (CCA). If the current signal strength
      * (RSSI/ED) is stronger than this CCA threshold value, the transceiver
-     * usually considers that the radio medium is busy. Otherwise, i.e.
-     * if RSSI/ED value is less than the CCA threshold value, the radio
-     * medium is supposed to be free (the possibly received weak signal
-     * is considered to be background, meaningless noise).
+     * usually considers that the radio medium is busy. Otherwise, i.e. if
+     * RSSI/ED value is less than the CCA threshold value, the radio medium is
+     * supposed to be free (the possibly received weak signal is considered to
+     * be background, meaningless noise).
      *
-     * Most transceivers allow to set this CCA threshold value.
-     * Some research work has proven that dynamically adapting it
-     * to network environment can improve QoS, especially in WSN.
+     * Most transceivers allow to set this CCA threshold value. Some research
+     * work has proven that dynamically adapting it to network environment can
+     * improve QoS, especially in WSN.
      */
     NETOPT_CCA_THRESHOLD,
 
     /**
-     * @brief CCA mode for the radio transceiver
+     * @brief   (uint8_t) CCA mode for the radio transceiver
      *
-     * Get/set the CCA mode as uint8_t
-     * corresponding to the respective PHY standard.
+     * Get/set the CCA mode corresponding to the respective PHY standard.
      * - IEEE 802.15.4: @ref netdev_ieee802154_cca_mode_t
      */
     NETOPT_CCA_MODE,
 
     /**
-     * @brief get statistics about sent and received packets and data of the device or protocol
+     * @brief   (@ref netstats_t*) get statistics about sent and received packets and data of the device or protocol
      *
      * Expects a pointer to a @ref netstats_t struct that will be pointed to
      * the corresponding @ref netstats_t of the module.
      */
     NETOPT_STATS,
 
     /**
-     * @brief en/disable encryption.
+     * @brief   (@ref netopt_enable_t) link layer encryption.
      */
-    NETOPT_ENCRYPTION,        /**< en/disable encryption */
-    NETOPT_ENCRYPTION_KEY,    /**< set encryption key */
+    NETOPT_ENCRYPTION,
 
     /**
-     * @brief Test mode for the radio, e.g. for CE or FCC certification
+     * @brief   (byte array) set encryption key
```

Comment:
I'm not sure if this is in host or network byte order, but network byte order makes more sense for me here.

Reply:
Does an encryption key have a byte order, or is it simply regarded as an array of bytes?
e.g. do implementations actually reverse 128 bit AES keys when distributing them or are the algorithm specifications written to treat the keys as byte blobs when implementing?

Reply:
I honestly don't know, so I guess it is best to keep it unspecified for now.